compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1118,53 +1123,54 @@ private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, STNode visibilityQual
      * @param isObjectTypeDesc Whether object type or not
      * @return Parsed node
      */
-    private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, STNode visibilityQualifier,
+    private STNode parseFuncDefOrFuncTypeDescRhs(STNode metadata, STNode visibilityQualifier,
                                               List<STNode> qualifiers, STNode functionKeyword, boolean isObjectMember,
-                                              boolean isObjectTypeDesc) {
-        switch (peek(2).kind) {
+                                              boolean isObjectTypeDesc, STNode name) {
```

Comment:
We include booleans at the last of param list

Reply:
Fixed.